src/subsets.c

```diff
@@ -584,14 +584,35 @@ int do_subsets_crack(struct db_main *db, char *req_charset)
 		keyspace += numwords(i, charcount, subsets_cur_len, required);
 
 	if (john_main_process) {
+		int len;
+		uint64_t total_keyspace = 0;
+
+		for (len = subsets_cur_len; len <= maxlength; len++) {
+			for (i = min_comb; i <= MIN(len, maxdiff); i++) {
+				uint64_t nw = numwords(i, charcount, len, required);
+				if (total_keyspace + nw < total_keyspace) {
+					total_keyspace = 0;
+					break;
+				}
+				total_keyspace += nw;
```

Comment:
This can be simplified to first doing the `total_keyspace += nw;` and then seeing if `total_keyspace < nw`. So that you don't need to perform the addition twice.